partner_communication/models/mail_template.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Joel Vaucher <jvaucher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import api, models
+
+
+class MailTemplate(models.Model):
+    """change generate_email to can be call with res_ids
+    not only an id but with a record, this is use for the change
+    of email_preview.template res_id selection to reference field"""
+    _inherit = 'mail.template'
+
+    @api.multi
+    def generate_email(self, res_ids, fields=None):
+
+        if isinstance(res_ids, models.BaseModel):
+            res_ids = res_ids.id
```

Comment:
wouldn't it be `res_ids.ids`? Also I'm not sure why you need this? Should you rather fix the caller if he calls the method with wrong parameters?

Reply:
It's because the caller is a method in the odoo source code

odoo/addons/mail/models/mail_template.py
method send_mail

Reply:
also the generate_email super method directly call
self.ensure_one()
so I didn't put it again here, should I ?